### ISSUE TITLE
lifecycleExperimental not works in react native

### ISSUE THREAD

**Issue by jose920405** [airbnb/enzyme#1279]
Example code:

**_In my test file:_**

```
shallow(
    <LoginScreen />,
    { context: { store }, lifecycleExperimental: true },
  );
```

**_In my component:_**

```
class LoginScreen extends Component {

  constructor(props) {
    super(props)
  }

  componentDidMount() {
    console.log('componentDidMount ===>')  // This log is never fired
  }

  componentDidUpdate() {
    console.log('componentDidUpdate ====> ') // This log is never fired
  }

  render() {
    console.log('Inside component Render') // This log is fired
    return (
      <View style={styles.mainContainer}>
      </View>
    )
  }
}
```

**Comment by ljharb** [comment]
What version of enzyme are you using? v3 doesn't have `lifecycleExperimental` anymore; it's on my default.

**Comment by jose920405** [comment]
hi @ljharb 

My version is `2.9.1`.

![captura de pantalla 2017-10-18 a la s 4 33 57 p m](https://user-images.githubusercontent.com/12536184/31743826-31a88770-b422-11e7-9ce2-bfec4ee8ee47.png)

**Comment by jose920405** [comment]
My react info:

```
"react": "16.0.0-alpha.12",
"react-native": "0.47.2",
```

I just realized that there are [problems with version 16 alpha of react](https://github.com/airbnb/enzyme/issues/928). This is the reason for the problem I am having

**Comment by ljharb** [comment]
@jose920405 since real v16 is out, can you try that?

**Comment by jose920405** [comment]
ok. i'll try and i supose that is necessary use `enzyme-adapter-react-16` so i will instal it too.

**Comment by jose920405** [comment]
@ljharb 

Yeah, update my react version to `16.0.0` and use `enzyme-adapter-react-16` fix my problems. Thanks

**Comment by jose920405** [comment]
hi @ljharb,

I'm testing a lot more thoroughly and I can see that `componentDidUpdate()` is not running. Is there a rule for this?

The render runs correctly each time, but the componentDidUpdate does not. I'm clear that I'm not doing anything special to try to get this to run, I'm just triggering a `press` and letting my code work on its own

```
const buttons = render.find('TouchableOpacity');
const loginButton = buttons.first()
loginButton.simulate('press')
```

**Comment by jose920405** [comment]
@ljharb

I am testing and apparently this is a problem related to redux interactions. When I change my redux `dispatch` by a simple `setState`, `componentDidUpdate` is executed correctly
